python/semantic_kernel/kernel_extensions/kernel_filters_extension.py

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+from functools import partial
+from typing import Any, Callable, Coroutine, Literal, TypeVar
+
+from pydantic import Field
+
+from semantic_kernel.filters.filter_context_base import FilterContextBase
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+
+FILTER_CONTEXT_TYPE = TypeVar("FILTER_CONTEXT_TYPE", bound=FilterContextBase)
+CALLABLE_FILTER_TYPE = Callable[[FILTER_CONTEXT_TYPE, Callable[[FILTER_CONTEXT_TYPE], None]], None]
+
+
+ALLOWED_FILTERS = ["function_invocation", "prompt_rendering", "auto_function_invocation"]
+ALLOWED_FILTERS_LITERAL = Literal["function_invocation", "prompt_rendering", "auto_function_invocation"]
+FILTER_MAPPING = {
+    "function_invocation": "function_invocation_filters",
+    "prompt_rendering": "prompt_rendering_filters",
+    "auto_function_invocation": "auto_function_invocation_filters",
+}
+
+
+class KernelFilterExtension(KernelBaseModel):
+    """KernelFilterExtension."""
+
+    function_invocation_filters: list[tuple[int, CALLABLE_FILTER_TYPE]] = Field(default_factory=list)
+    prompt_rendering_filters: list[tuple[int, CALLABLE_FILTER_TYPE]] = Field(default_factory=list)
+    auto_function_invocation_filters: list[tuple[int, CALLABLE_FILTER_TYPE]] = Field(default_factory=list)
+
+    def add_filter(self, filter_type: ALLOWED_FILTERS_LITERAL, filter: CALLABLE_FILTER_TYPE) -> None:
+        """Add a filter to the Kernel.
+
+        Each filter is added to the beginning of the list of filters,
+        this is because the filters are executed in the order they are added,
+        so the first filter added, will be the first to be executed,
+        but it will also be the last executed for the part after `await next(context)`.
+
+        Args:
+            filter_type (str): The type of the filter to add (function_invocation, prompt_rendering)
+            filter (object): The filter to add
+
+        """
+        if filter_type not in ALLOWED_FILTERS:
+            raise ValueError(f"filter_type must be one of {ALLOWED_FILTERS}")
```

Comment:
Should this be a new type like a `KernelFilterException` or `KernelFilterInitializationException`? Same question for the other ValueError exceptions in this file.